torch/_meta_registrations.py

```diff
@@ -36,3 +43,26 @@ def meta_max(self):
 @torch.library.impl(meta_lib, "min")
 def meta_min(self):
     return self.new_empty(())
+
+def squareCheckInputs(self, f_name):
+    assert self.dim() >= 2, f"{f_name}: The input tensor must have at least 2 dimensions."
+    # TODO: I think the error message has the -2 and -1 swapped.  If you fix
+    # it fix the C++ squareCheckInputs too
+    assert self.size(-1) == self.size(-2), \
+        f"{f_name}: A must be batches of square matrices, but they are {self.size(-1)} by {self.size(-2)} matrices"
+
+def checkUplo(uplo: str):
+    uplo_uppercase = uplo.upper()
+    assert len(uplo) == 1 and uplo_uppercase == 'U' or uplo_uppercase == 'L', \
+        f"Expected UPLO argument to be 'L' or 'U', but got {uplo}"
+
+@torch.library.impl(meta_lib, "linalg_eigh")
+def meta_linalg_eigh(self, uplo="L"):
```

Comment:
This is simpler than the corresponding C++ `out` function because it doesnt have to do all of the error checking of the passed in `values`/`vectors`... Which makes me wonder, are we ever going to do `FakeTensor/MetaTensor` for out variants or is that out of scope?

Reply:
The PrimTorch refs handle out checking. We should make our shape functions handle out variants but there's a trick: we can run the functional first, and then use that to determine what necessary error tests apply to out. This won't be parity, but it will be better than eager mode